tests/diagnostics/uninitialized-fields.slang

```diff
@@ -0,0 +1,92 @@
+//TEST:SIMPLE(filecheck=CHK): -target spirv
+
+extern static const int flag;
+
+struct None
+{
+    int a;
+    int b;
+
+    __init()
+    {
+        a = 0;
+        b = 0;
+    }
+}
+
+struct Simple
+{
+    int value;
+    Simple *next;
+
+    __init(int v, Simple* ptr)
+    {
+        value = v;
+        //CHK-DAG: warning 41020: exiting constructor without initializing field 'next'
+    }
+}
+
+struct Conditional
+{
+    int field;
+    int another_field;
+
+    __init()
+    {
+        field = 0;
+        if (flag == 0)
+            //CHK-DAG: warning 41020: exiting constructor without initializing field 'another_field'
+            return;
+
+        another_field = 0;
+    }
```

Comment:
I am not sure if a return is inserted at the end of the constructor.
Maybe we should have another case to test it like this,
```
    __init()
    {
        field = 0;
        if (flag == 0)
        {
            another_field = 0;
            return;
        }

        //CHK-DAG: warning 41020: exiting constructor without initializing field 'another_field'
    }
```

Reply:
I am not sure if the following case will print a warning,
```
    __init()
    {
        field = 0;
        if (flag == 0)
        {
            another_field = 0;
        }

        return;
    }
```

Reply:
It seems that constructors will always end with a return, so in the first case there will be a warning that is printed. The second case does not issue a warning because we do not checking for potentially uninitialized values. We only trigger warnings if it a field will *never* be initialized by the time of next use/return;